components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/handler/RegistrationHandlerTest.java

```diff
@@ -0,0 +1,89 @@
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.dcr.handler;
+
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oauth.dcr.context.DCRMessageContext;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequest;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequestProfile;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationResponse;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationResponseProfile;
+import org.wso2.carbon.identity.oauth.dcr.service.DCRManagementService;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+
+@PrepareForTest({DCRManagementService.class, RegistrationHandler.class})
+public class RegistrationHandlerTest extends PowerMockTestCase {
+
+    private RegistrationHandler registrationHandler;
+    private String testTenantDomain = "testTenantDomain";
+
+    @Mock
+    RegistrationRequest mockRegisterRequest;
+
+    @Mock
+    DCRMessageContext mockDcrMessageContext;
+
+    @Mock
+    RegistrationRequestProfile mockRegistrationRequestProfile;
+
+    @Mock
+    RegistrationResponseProfile mockRegistrationResponseProfile;
+
+    @Mock
+    DCRManagementService mockDCRManagementService;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        initMocks(this);
+        registrationHandler = new RegistrationHandler();
+    }
+
+    @Test
+    public void testHandle() throws Exception {
+        when(mockDcrMessageContext.getIdentityRequest()).thenReturn(mockRegisterRequest);
+        when(mockRegisterRequest.getRegistrationRequestProfile()).thenReturn(mockRegistrationRequestProfile);
+        when(mockRegisterRequest.getTenantDomain()).thenReturn(testTenantDomain);
+
+        mockStatic(DCRManagementService.class);
+        when (DCRManagementService.getInstance()).thenReturn(mockDCRManagementService);
+        when(mockDCRManagementService.registerOAuthApplication(mockRegistrationRequestProfile)).
+                thenReturn(mockRegistrationResponseProfile);
+
+        RegistrationResponse.DCRRegisterResponseBuilder registerResponseBuilder =
+                new RegistrationResponse.DCRRegisterResponseBuilder();
+        whenNew(RegistrationResponse.DCRRegisterResponseBuilder.class).withNoArguments().thenReturn(registerResponseBuilder);
+
+        assertEquals(registrationHandler.handle(mockDcrMessageContext), registerResponseBuilder);
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
```

Comment:
you don't need this method declaration since you have already extend **PowerMockTestCase**